jcl/src/java.base/share/classes/java/lang/Class.java

```diff
@@ -4586,26 +4586,34 @@ public boolean isNestmateOf(Class<?> that) {
  */
 @CallerSensitive
 public Class<?>[] getNestMembers() throws LinkageError, SecurityException {
+	Class<?>[] result;
+	
 	if (isArray() || isPrimitive()) {
 		/* By spec, Class objects representing array types or primitive types
 		 * belong to the nest consisting only of itself.
 		 */
-		return new Class<?>[] { this };
-	}
-	SecurityManager securityManager = System.getSecurityManager();
-	if (securityManager != null) {
-		/* All classes in a nest must be in the same runtime package and therefore same classloader */
-		ClassLoader nestMemberClassLoader = this.internalGetClassLoader();
-		ClassLoader callerClassLoader = ClassLoader.getCallerClassLoader();
-		if (!doesClassLoaderDescendFrom(nestMemberClassLoader, callerClassLoader)) {
-			String nestMemberPackageName = this.getPackageName();
-			if ((nestMemberPackageName != null) && (nestMemberPackageName != "")) {
-				securityManager.checkPackageAccess(nestMemberPackageName);
+		result = new Class<?>[] { this };
+	} else {
+		result = getNestMembersImpl();
+		if (result.length == 1 && result[0] == this) {
+			/* This is a Class object that belongs to the nest consisting only of itself */
```

Comment:
For java methods we don't enforce single point of return.  

I think this is clearer if the code is ordered:
```java
if (isArray() || isPrimitive()) {
		/* By spec, Class objects representing array types or primitive types
		 * belong to the nest consisting only of itself.
		 */
		return new Class<?>[] { this };
}
Class<?>[] members = getNestMembersImpl();
if (members.length > 1) {
  // security check here
}
return members;
```

Reply:
Updated PR, please have another look.

P.S. I have to share part of blame (missed this singleton case during reviewing).